ibm-ups/ups.hpp

```diff
@@ -108,6 +108,49 @@ class UPS : public DeviceObject
      */
     void closeDevice();
 
+    /**
+     * Force a PropertiesChanged event to be emitted for the BatteryLevel
+     * property.
+     */
+    void emitBatteryLevelChangedEvent()
+    {
+        // Save current value, change property without emitting a signal, and
+        // then change back to the saved value.  There is no direct way to force
+        // the signal to be emitted using the sdbusplus API.
```

Comment:
_serverInterface.property_changed(property);

Reply:
The problem I ran into is that the sdbusplus::server::interface::interface data member (_serverInterface) is private.

Here is the C++ server class generated based on the UPower Device interface YAML:

```
class Device
{
...
    private:
...
        sdbusplus::server::interface::interface _org_freedesktop_UPower_Device_interface;
...
}
```

Reply:
Ah, in my usage I'm not generating from YAML and have my own server interface instance variable. Generated code is evil and wrong.